pages/sign_up.tsx

```diff
@@ -49,9 +52,19 @@ export async function getStaticProps({ locale }: any) {
 
 const SignUp: NextPageWithLayout = () => {
   const { signup, login, register } = useAuth();
-  const { getItem } = useStorage();
+  const { setItem, getItem } = useStorage();
   const router = useRouter();
   const { t } = useTranslation("signUpProps");
+  const [claimPerson] = useMutation(CLAIM_DID);
+
+  const claim = async (id: string) => {
+    try {
+      const { data } = await claimPerson({ variables: { id } });
+      setItem("didId", data?.claimPerson.did.result.didDocument.id);
+    } catch (err) {
+      devLog("signUp", "claim", err);
+    }
+  };
```

Comment:
Forse anche qui potresti fare semplicemente

```tsx
const { data } = useMutation(CLAIM_DID, { variables: { id } });
setItem("didId", data?.claimPerson.did.result.didDocument.id);
```

Reply:
ad esempio in questo caso {data} é il valore di ritorno della mutate function (la chiama cosí la documentazione di apollo client